packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Collection, Node} from '@react-stately/collections';
+import {CollectionBase, SingleSelection} from '@react-types/shared';
+import {FocusStrategy} from '@react-types/menu';
+import {Key, useEffect, useMemo} from 'react';
+import {ListState} from '@react-stately/list';
+import {useCollator} from '@react-aria/i18n';
+import {useControlledState} from '@react-stately/utils';
+import {useSelectState} from '@react-stately/select';
+
+// Can't extend SelectState because we modify toggle/open here, copied types for now
+export interface ComboBoxState<T> extends ListState<T> {
+  /** The key for the currently selected item. */
+  selectedKey: Key,
+  /** Sets the selected key. */
+  setSelectedKey: (key: Key) => void,
+  /** The value of the currently selected item. */
+  selectedItem: Node<T>,
+  /** Whether the select is currently focused. */
+  isFocused: boolean,
+  /** Sets whether the select is focused. */
+  setFocused: (isFocused: boolean) => void,
+  /** Whether the menu is currently open. */
+  isOpen: boolean,
+  /** Sets whether the menu is open. */
+  setOpen(value: boolean): void,
+  /** Controls which item will be auto focused when the menu opens. */
+  focusStrategy: FocusStrategy,
+  /** Sets which item will be auto focused when the menu opens. */
+  setFocusStrategy(value: FocusStrategy): void,
```

Comment:
A bunch of the above are already in the `SelectState` type. You could probably extend from that?